linkerd/core/src/main/scala/io/buoyant/linkerd/RequestEvaluator.scala

```diff
@@ -0,0 +1,168 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.http._
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util._
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RouterLabel
+import io.buoyant.router.RoutingFactory
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+
+class RequestEvaluator(
+  endpoint: EndpointAddr,
+  namer: DstBindingFactory.Namer,
+  dtab: BaseDtab,
+  label: String
+) extends SimpleFilter[Request, Response] {
+
+  private[this] val RequestTracerMaxDepthHeader = "l5d-max-depth"
+
+  private[this] def printEvaluatedRequest(
+    prependText: String,
+    routerLabel: String,
+    serviceName: String,
+    clientName: String,
+    selectedAddress: String,
+    addresses: Option[Set[String]],
+    dtabResolution: List[String]
+  ) = {
+    Seq(
+      prependText,
+      s"""
+         |--- Router: $routerLabel ---
+         |service name: $serviceName
+         |client name: $clientName
+         |selected address: $selectedAddress
+         |addresses: [${addresses.getOrElse(Set.empty).mkString(", ")}]
+         |dtab resolution:
+         |${dtabResolution.map("  " + _).mkString("\n")}
+    """.stripMargin
+    ).mkString("\n").trim
+  }
+
+  private[this] def formatDTree(
+    dTree: DelegateTree[Name.Bound],
+    tree: List[String],
+    searchPath: Path
+  ): List[String] = {
+    dTree match {
+      case Leaf(path, _, bound) if bound.id == searchPath =>
+        tree :+ path.show
+      case Transformation(_, _, value, remainingTree) =>
+        formatDTree(remainingTree, tree :+ value.path.show, searchPath)
+      case Delegate(path, _, remainingTree) =>
+        formatDTree(remainingTree, tree :+ path.show, searchPath)
+      case Union(path, _, remainingWeightedTrees@_*) =>
+        remainingWeightedTrees.map { wd =>
+          formatDTree(wd.tree, tree :+ s"${wd.weight} * ${path.show}", searchPath)
```

Comment:
I think this is not quite right.  Technically the weight belongs to the subtree (wd.tree) instead of this node (path).  This causes delegations like 
```
  /svc/default
  1.0 * /x/default
  /b/default
  /$/inet/127.1/4140
```
when really it should be

```
  /svc/default
  /x/default
  1.0 * /b/default
  /$/inet/127.1/4140
```

Regardless, I'm not sure if it's even valuable to print the weight at all.  We can probably just omit it.